test/controllers/passwordless/sessions_controller_test.rb

```diff
@@ -47,7 +47,19 @@ def create_session_for(user)
 
       assert_equal 200, status
       assert_equal '/', path
-      refute_nil cookies[:user_id]
+      assert_not_nil cookies[:user_id]
+    end
+
+    test 'signing in via a token as STI model' do
+      admin = Admin.create email: 'a@a'
+      session = create_session_for admin
+
+      get "/users/sign_in/#{session.token}"
+      follow_redirect!
+
+      assert_equal 200, status
+      assert_equal '/', path
+      assert_not_nil cookies[:user_id]
```

Comment:
Thanks for writing the test! I meant to add it, but didn't get around to it in time 😄